server/paymentProviders/utils.js

```diff
@@ -235,15 +236,20 @@ export const persistTransaction = async (virtualCard, transaction) => {
 
 export const getOrCreateVendor = async (vendorProviderId, vendorName) => {
   const slug = vendorProviderId.toString().toLowerCase();
+  const uniqueSlug = crypto.hash(`${slug}${vendorName}`);
```

Comment:
Could we try to have slugs that are a bit more human friendly such as:

`slugify(vendorName)-whateverHash`

Reply:
We can. But keep in mind that we do not render or expose vendor slugs anywhere.